base/compiler/abstractinterpretation.jl

```diff
@@ -95,7 +95,8 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
     if f !== nothing && napplicable == 1 && is_method_pure(applicable[1]::MethodMatch)
         val = pure_eval_call(f, argtypes)
         if val !== false
-            return CallMeta(val, info)
+            # TODO: add some sort of edge(s)
+            return CallMeta(val, MethodResultPure())
```

Comment:
Other future consumers of stmt info (e.g. Cthulhu) may want to look at the `info` here, probably, but since that's not ready yet, I think we can cross that bridge when we get there.

Reply:
Sure, but when we hit a call we think might be pure because of `info`, we're then ignoring the actual `info` when we do the `pure_eval_call`, so I think this is actually a bit more accurate.